src/chain/store/index.rs

```diff
@@ -143,16 +143,12 @@ impl<BS: Blockstore> ChainIndex<BS> {
             return self.walk_back(from, to);
         }
         let total_size = from.epoch() - to;
-        let pb = ProgressBar::new(total_size as u64);
-        pb.message("Scanning blockchain ");
-        pb.set_max_refresh_rate(Some(std::time::Duration::from_millis(500)));
+        let pl = ProgressLog::new("Scanning blockchain", total_size as u64);
```

Comment:
Now that I'm thinking about this, why not do something like:

```
ProgressLog::new().with_message("Scanning blockchain").with_total_size(total_size)
```

Makes it more flexible and explicit.

Reply:
So like some builder pattern? I did one but with @aatifsyed we agree it was superfluous.

Reply:
Well, this won't require the actual builder. No strong opinion here, however I don't see a reason to require `total_size` at construction, it does not always work. What that means is we'd have to explicitly pass 0 there, which makes the interface confusing and a bit dirty.

Reply:
We always have a `total_size`, don't we?

Reply:
We do always have it.